soroban-env-host/src/native_contract/token/contract.rs

```diff
@@ -297,6 +301,14 @@ impl TokenTrait for Token {
         Ok(())
     }
 
+    fn is_admin(e: &Host, address: Address) -> Result<bool, HostError> {
+        let current_admin = read_administrator(e);
+        match current_admin {
+            Ok(admin) => return Ok(e.compare(&admin, &address)? == Ordering::Equal),
+            Err(_) => Ok(false),
```

Comment:
This is wrong, isn't it? We should never suppress the errors (here it means something like bad footprint or running out of budget etc.). We should also always have an admin. So this can be changed to `Ok(e.compare(&read_administrator(e)?, &address)?.is_eq())` or just `read_administrator(e)` if we opt to just use `admin` function as suggested above.

Reply:
Ah yeah good point

Reply:
Well the lumen contract doesn't have an admin. We should still return false in that case and not error.